memcache/memcache.go

```diff
@@ -29,6 +29,8 @@ import (
 	"strings"
 	"sync"
 	"time"
+
+	"github.com/thanos-io/thanos/pkg/pool"
```

Comment:
This is supposed to be a minimal, low-level, library. It cannot depend on a massive application.

However, the Go idiom is to declare interfaces where they are used, not import them from some other place. So just copy-paste the methods you need and make a new interface here.